### ISSUE TITLE
Question about the code

### ISSUE THREAD

**Issue by yaxum** [zake7749/DeepToxic#5]
Hello!

I am new to Python and NLP. Although I am missing input variables in the function "train_folds(train_features, test_features) " or have I missed something?

Sincerely,
Yaxum Cedeno

**Comment by zake7749** [comment]
Hi yaxum,

I guess you were talking about `LogisticRegression.ipynb`. This notebook was for doing experiments at the beginning of the competition and it actually has nothing to do with this project.

The function `train_folds` and `_train_model` were deprecated. I would recommend skipping them and starting from cells under the title `Logistic Regression`. If you are curious about the implementation of LogisticRegression with CV, I refer you to this [clean and awesome kernel](https://www.kaggle.com/thousandvoices/logistic-regression-with-words-and-char-n-grams).

All the best,
Justin Yang

**Comment by yaxum** [comment]
Hi Zake!

Thank you for the tips!
I have had a hard time implementing a NLP process on multiple articles with multiple labels. Do you have any tips on articles or other codes that I can get inspiration from? Besides the link you handed to me now.


Sincerely,
Yaxum Cedeno

________________________________

**Comment by zake7749** [comment]
Sure. I would suggest reading [[For Beginners] Tackling Toxic Using Keras](https://www.kaggle.com/sbongo/for-beginners-tackling-toxic-using-keras) and [Deep Learning for NLP with Pytorch](https://pytorch.org/tutorials/beginner/deep_learning_nlp_tutorial.html).

**Comment by yaxum** [comment]
Thank you very much!
This helped me alot.

I took the example from https://www.kaggle.com/thousandvoices/logistic-regression-with-words-and-char-n-grams and tried it out on my data. I have multiple articles with multiple labels (industry, informatics, .... ) after 3 hours it was done the model however was the result in many different decimals of the probability of every row for every label. However did I though that the output would be a binomial output with either 0 or 1 per row and label. What am I missing and thinking wrong?

Sincerely,
Yaxum Cedeno


________________________________

**Comment by zake7749** [comment]
The outputs are correct since this task is for **multi-label** classification rather than **multi-class** classification.

**Comment by yaxum** [comment]
Sorry I expressed myself wrong. I Supposed that the output would be either 0 or 1 not a probability. Although I did do my own cut off with the number 0.5 to replace all numbers below 0.5 with 0 and above with 1. Thank you again.

________________________________
